app/routes/analysis/analysis.py

```diff
@@ -87,61 +76,42 @@ async def zonal_statistics_post(request: ZonalAnalysisRequestIn):
 
 
 async def _zonal_statistics(
-    geometry: Geometry,
+    geometry: Dict[str, Any],
     sum_layers: List[RasterLayer],
     group_by: List[RasterLayer],
     filters: List[RasterLayer],
     start_date: Optional[str],
     end_date: Optional[str],
 ):
-    payload = {
-        "geometry": jsonable_encoder(geometry),
-        "group_by": group_by,
-        "filters": filters,
-        "sum": sum_layers,
-        "start_date": start_date,
-        "end_date": end_date,
-    }
-
-    try:
-        response = await _invoke_lambda(payload)
-    except httpx.TimeoutException:
-        raise HTTPException(500, "Query took too long to process.")
-
-    try:
-        response_data = response.json()["body"]["data"]
-    except (JSONDecodeError, KeyError):
-        logger.error(
-            f"Raster analysis lambda experienced an error. Full response: {response.text}"
-        )
-        raise HTTPException(
-            500, "Raster analysis geoprocessor experienced an error. See logs."
-        )
-
-    return Response(data=response_data)
-
-
-async def _invoke_lambda(payload, timeout=55) -> httpx.Response:
-    session = boto3.Session()
-    cred = session.get_credentials()
-
-    aws = AWS4Auth(
-        access_id=cred.access_key,
-        secret_key=cred.secret_key,
-        security_token=cred.token,
-        region=AWS_REGION,
-        service="lambda",
-    )
+    base = sum_layers[0].value
+    selectors = ",".join([f"sum({lyr.value})" for lyr in sum_layers])
+    groups = ",".join([lyr.value for lyr in group_by])
+
+    where_clauses = []
+    for lyr in filters:
+        if "umd_tree_cover_density" in lyr.value:
+            where_clauses.append(f"{lyr.value[:-2]}__threshold = {lyr.value[-2:]}")
+        else:
+            where_clauses.append(f"{lyr.value} != 0")
+
+    if start_date:
+        where_clauses.append(_get_date_filter(start_date, ">="))
+
+    if end_date:
+        where_clauses.append(_get_date_filter(end_date, "<"))
+
+    where = " and ".join(filters)
+
+    query = f"select {selectors} from {base}"
+    if where:
+        query += f" where {where}"
+    query += f"group by {groups}"
```

Comment:
OK, but I would prefer to stay consistent though out the code base and use SQL Alchemy core to compose SQL queries.

Here some examples in use:

https://github.com/wri/gfw-data-api/blob/master/app/crud/geostore.py#L32
https://github.com/wri/gfw-data-api/blob/master/batch/python/export_1x1_grid.py#L363-L375

Reply:
@thomas-maschler ok I can do that, is the end result a SQL string or some object specific to sqlachemy?

Reply:
the string representation will be a SQL expression ie : `str(query)`